src/config.rs

```diff
@@ -0,0 +1,99 @@
+//! Configuration for `HttpClient`s.
+
+/// Configuration for `HttpClient`s.
+#[non_exhaustive]
+#[derive(Clone, Debug)]
+pub struct Config {
+    /// TCP `NO_DELAY`.
+    pub no_delay: Option<Option<bool>>,
```

Comment:
I'm curious why this needs to be `Option<Option<bool>>` instead of just `Option<bool>`. What does the extra option communicate? "they haven't set this setting to say that they don't want to set the setting?"

Reply:
Combined with the comment below, this is `Option<Config { no_delay: Option<Option<bool>> }>`, which could just be `Config { no_delay: Option<bool> }` — either it's true, or false, or they haven't set it.

Reply:
The idea is for updating config at a later time, this way you can set which options are being changed and which aren't.

The outer option being "change this value or not", and the inner being "use a default value or not".

This assumes that updating after client creation is actually a useful thing that people would want to do. I admit the actual implementation of that would probably not be super trivial.